source/PlayerInfo.cpp

```diff
@@ -1782,36 +1783,88 @@ void PlayerInfo::HandleEvent(const ShipEvent &event, UI *ui)
 // Get the value of the given condition (default 0).
 int64_t PlayerInfo::GetCondition(const string &name) const
 {
-	auto it = conditions.find(name);
-	return (it == conditions.end()) ? 0 : it->second;
+	return conditions.GetCondition(name);
+}
+
+
+
+// Set a condition to the given value.
+bool PlayerInfo::SetCondition(const string &name, int64_t value)
+{
+	return conditions.SetCondition(name, value);
+}
+
+
+
+// Add a value to the given condition.
+bool PlayerInfo::AddCondition(const string &name, int64_t value)
+{
+	return conditions.AddCondition(name, value);
+}
+
+
+
+// Remove the value for some condition.
+bool PlayerInfo::EraseCondition(const string &name)
+{
+	return conditions.EraseCondition(name);
+}
+
+
+
+// Temporary function to erase conditions by prefix. This function should
+// no longer be needed once automatically generated conditions are on-demand
+// automatically generated.
+// This function is performance-wise also not very efficient, but that
+// might be okay for a temporary function.
+void PlayerInfo::EraseManualByPrefix(const string &prefix)
+{
+	set<string> toErase;
```

Comment:
Why can't we erase as we go? Wouldn't you just need to walk back the iterator when something matches and gets erased.

Reply:
The list of manual conditions is only provided as `const` (meaning read-only). I could make it writeable as well, but I'm afraid that that could lead to bugs in the future if developers don't notice that they are working with a subset of the conditions.